### ISSUE TITLE
Add `vector.contains(v1, v2, pos)` util function

### ISSUE THREAD

**Issue by Zughy** [minetest/minetest#13366]
## Problem
There is no util function to check whether a position is inside between a cuboid area, unless by creating a VoxelArea first

## Solutions
`vector.contains(v1, v2, pos)`

**Comment by fluxionary** [comment]
checking whether two regions intersect is also a similar and really common function

**Comment by jeremyshannon** [comment]
Yeah I had to write my own function for this, but it really should be in the API. It seems like a no-brainer.

**Comment by Desour** [comment]
For some reason I thought there was already a feature request for this, but I can't find anything.

:+1: Last time I had to check whether a pos in an area, it was super annoying.

There should be 2 versions:
* `vector.containsi(v1, v2, p)`: checks `v1 <= p <= v2` ("i" for "inclusive" or "integral")
* `vector.contains(v1, v2, p)` or `vector.containse(v1, v2, p)`: checks `v1 < p < v2` ("e" for exclusive)

Note:
There's already `VoxelArea(v1, v2):containsp(p)` that does the same as `vector.containsi(v1, v2, p)`. (The new helper is still justified in `vector`.)

**Comment by Zughy** [comment]
I think it's overengineered having 2 functions. From the docs (VoxelArea section):
> The coordinates are *inclusive*, like most other things in Minetest

Also, `containsi` and `contains` do different things in VoxelAreas than the ones you've suggested, it'd create confusion. I'd make `vector.contains` inclusive and that's it (I had expected it to return `true` even if `pos` is on one of the edges even before we had this conversation)
